mlir-sycl/lib/Conversion/SYCLToLLVM/SYCLToLLVM.cpp

```diff
@@ -2050,22 +2063,128 @@ void mlir::populateSYCLToLLVMConversionPatterns(
 
 namespace {
 /// A pass converting MLIR SYCL operations into LLVM dialect.
+///
+/// This pass relies on SYCL to GPU and target dialects, e.g., SPIRV, conversion
+/// patterns. This pass is executed in 5 steps:
+/// 1. Convert grid ops to a target dialect, e.g., SPIRV;
+/// 2. Lower all of the operations to LLVM. As some of the operations will yield
+///    SYCL grid ops, we need to perform the same conversions as in step 1
```

Comment:
What kind of operations will yield SYCL grid ops? 
Is is possible to lower to LLVM first, then to SPIRV, then LLVM again? (saving one invocation to SPIRV)

Reply:
opps, the comment needs to be updated.

Reply:
I'm super clumsy today :sweat_smile: